### PR TITLE
Update the config for assignment of community PRs

### LINKED ISSUE #34215: Assign pull request reviews to community contributions
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

**Issue Description:**

Community pull requests should be assigned a reviewer Automatically in an effort to foster empowerment amongst community developers. 

After https://github.com/woocommerce/woocommerce/issues/34197 is completed, create a Github action that checks to see if pull requests are community contributions, and if so, assign a reviewing team based on where the code changes lie.

### PR DESCRIPTION
**Pull request by jonathansadowski**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR updates the configuration for the community PR reviewer assignment. Once this is merged, any PRs in the glob path matching the top-level patterns will be assigned to the teams listed.

Admittedly, some of these assignments are arbitrary, I don't know the teams that are the best owners of each section of code, so if you have a better suggestion, let me know, and I can update it... if not, we can have the teams edit this config themselves once they experience an incorrect suggestion 😄 

Closes #34215 

### How to test the changes in this Pull Request:

1. Make sure that the team assignments make sense, and that the yaml doesn't have any syntax errors
2.
3.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you successfully run tests with your changes locally?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> run changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by psealock** [comment]
@adrianduffell and @AnnaMag I've added you as reviewers to this PR. Could you confirm the configs proposed in this PR? It designates which team to assign a pull request to when a pull request is opened from outside WooCommerce/a8c based on which part of the codebase is being changed. This is an ongoing effort to surface these types of community contributions and making sure they don't go unnoticed.

**Comment by adrianduffell** [comment]
Thanks @psealock, is it possible to assign both Ghidorah and Mothra to items? We have a shared maintenance responsibility on many of them.

**Comment by jonathansadowski** [comment]
> Thanks @psealock, is it possible to assign both Ghidorah and Mothra to items? We have a shared maintenance responsibility on many of them.

Yep, I have a P2 post queued up for once this is merged, to explain that this'll be in place, and that it can be edited. Given the low number of community contributions we have in place for some of those areas of code, I don't think it's essential that we have the config 100% when it's merged.

A shared config would look something like this:

```
"packages/js/components/**/*":
  - team: mothra
  - team: ghidorah
```

**Comment by AnnaMag** [comment]
+1 to @adrianduffell's request. Thank you!

File: .github/project-community-pr-assigner.yml
Comment (adrianduffell) [inline .github/project-community-pr-assigner.yml:25]:
```suggestion
  - team: mothra
  - team: ghidorah
```

File: .github/project-community-pr-assigner.yml
Comment (adrianduffell) [inline .github/project-community-pr-assigner.yml:38]:
```suggestion
  - team: mothra
  - team: ghidorah
```

File: .github/project-community-pr-assigner.yml
Comment (adrianduffell) [inline .github/project-community-pr-assigner.yml:54]:
```suggestion
  - team: mothra
  - team: ghidorah
```

File: .github/project-community-pr-assigner.yml
Comment (adrianduffell) [inline .github/project-community-pr-assigner.yml:73]:
```suggestion
  - team: mothra
  - team: ghidorah
```

File: .github/project-community-pr-assigner.yml
Comment (adrianduffell) [inline .github/project-community-pr-assigner.yml:77]:
```suggestion
  - team: mothra
  - team: ghidorah
```

File: .github/project-community-pr-assigner.yml
Comment (adrianduffell) [inline .github/project-community-pr-assigner.yml:81]:
```suggestion
  - team: mothra
  - team: ghidorah
```

**Review by adrianduffell — CHANGES_REQUESTED** [review]
Thanks for this, I've suggested adding Ghidorah to a few of the key areas we share with Mothra.

@amk, I think we could fine tune this later in a follow-up PR. What do you think?

**Review by samueljseay — DISMISSED** [review]
From my perspective these changes look good, so its just up to the various teams if they're happy with the assignments 👍🏻

File: .github/project-community-pr-assigner.yml
Comment (AnnaMag) [reply .github/project-community-pr-assigner.yml:81]: @adrianduffell sounds perfect, thank you.

**Review by AnnaMag — DISMISSED** [review]
LGTM, pending @adrianduffell comment.
Thank you.

**Comment by jonathansadowski** [comment]
Thanks @AnnaMag and @adrianduffell for confirming the assignments and suggesting changes. I've committed the ones that were suggested, and I'll publish a P2 shortly announcing this. We can see how well these work, and then any further changes can be made via future PRs!

**Pull request merged**
